<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from gripper.xacro                  | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="gripper" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <!-- Command to run:
  rosrun xacro xacro.py gripper.xacro > gripper.urdf; roslaunch urdf_tutorial display.launch model:=gripper.urdf gui:=True
  -->
  <!-- Materials -->
  <!-- Variables -->
  <!-- Finger macro -->
  <link name="base_link">
    <visual>
      <geometry>
        <box rpy="0 0 0" size="0.056 0.040 0.065"/>
      </geometry>
      <material name="Metal">
        <color rgba="0.8 0.78 0.83 1.0"/>
      </material>
    </visual>
  </link>
  <!-- Create 3 fingers -->
  <link name="f1_finger_base">
    <visual>
      <origin rpy="0 0 0" xyz="0 0.028 0"/>
      <geometry>
        <box rpy="0 0 0" size="0.007 0.056 0.014"/>
      </geometry>
      <material name="Darkmetal">
        <color rgba="0.5 0.49 0.56 1.0"/>
      </material>
    </visual>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.014" radius="0.0035"/>
      </geometry>
      <material name="Darkmetal">
        <color rgba="0.5 0.49 0.56 1.0"/>
      </material>
    </visual>
  </link>
  <link name="f1_finger_top">
    <visual>
      <origin rpy="0 0 0" xyz="0 0.023 0"/>
      <geometry>
        <box size="0.007 0.046 0.014"/>
      </geometry>
      <material name="Darkmetal">
        <color rgba="0.5 0.49 0.56 1.0"/>
      </material>
    </visual>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.014" radius="0.0035"/>
      </geometry>
      <material name="Darkmetal">
        <color rgba="0.5 0.49 0.56 1.0"/>
      </material>
    </visual>
  </link>
  <joint name="f1_finger_basejoint" type="revolute">
    <parent link="base_link"/>
    <child link="f1_finger_base"/>
    <origin rpy="0 0 0" xyz="0.028 0.02 0"/>
    <limit effort="1000.0" lower="-0.785" upper="0.087" velocity="0.5"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="f1_finger_topjoint" type="revolute">
    <parent link="f1_finger_base"/>
    <child link="f1_finger_top"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0 0 0" xyz="0 0.056 0"/>
    <limit effort="1000.0" lower="1.431" upper="0.489" velocity="0.5"/>
  </joint>
  <link name="f2_finger_base">
    <visual>
      <origin rpy="0 0 0" xyz="0 0.028 0"/>
      <geometry>
        <box rpy="0 0 0" size="0.007 0.056 0.014"/>
      </geometry>
      <material name="Darkmetal">
        <color rgba="0.5 0.49 0.56 1.0"/>
      </material>
    </visual>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.014" radius="0.0035"/>
      </geometry>
      <material name="Darkmetal">
        <color rgba="0.5 0.49 0.56 1.0"/>
      </material>
    </visual>
  </link>
  <link name="f2_finger_top">
    <visual>
      <origin rpy="0 0 0" xyz="0 0.023 0"/>
      <geometry>
        <box size="0.007 0.046 0.014"/>
      </geometry>
      <material name="Darkmetal">
        <color rgba="0.5 0.49 0.56 1.0"/>
      </material>
    </visual>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.014" radius="0.0035"/>
      </geometry>
      <material name="Darkmetal">
        <color rgba="0.5 0.49 0.56 1.0"/>
      </material>
    </visual>
  </link>
  <joint name="f2_finger_basejoint" type="revolute">
    <parent link="base_link"/>
    <child link="f2_finger_base"/>
    <origin rpy="0 3.14 0" xyz="-0.028 0.02 0.019"/>
    <limit effort="1000.0" lower="-0.785" upper="0.087" velocity="0.5"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="f2_finger_topjoint" type="revolute">
    <parent link="f2_finger_base"/>
    <child link="f2_finger_top"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0 0 0" xyz="0 0.056 0"/>
    <limit effort="1000.0" lower="1.431" upper="0.489" velocity="0.5"/>
  </joint>
  <link name="f3_finger_base">
    <visual>
      <origin rpy="0 0 0" xyz="0 0.028 0"/>
      <geometry>
        <box rpy="0 0 0" size="0.007 0.056 0.014"/>
      </geometry>
      <material name="Darkmetal">
        <color rgba="0.5 0.49 0.56 1.0"/>
      </material>
    </visual>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.014" radius="0.0035"/>
      </geometry>
      <material name="Darkmetal">
        <color rgba="0.5 0.49 0.56 1.0"/>
      </material>
    </visual>
  </link>
  <link name="f3_finger_top">
    <visual>
      <origin rpy="0 0 0" xyz="0 0.023 0"/>
      <geometry>
        <box size="0.007 0.046 0.014"/>
      </geometry>
      <material name="Darkmetal">
        <color rgba="0.5 0.49 0.56 1.0"/>
      </material>
    </visual>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.014" radius="0.0035"/>
      </geometry>
      <material name="Darkmetal">
        <color rgba="0.5 0.49 0.56 1.0"/>
      </material>
    </visual>
  </link>
  <joint name="f3_finger_basejoint" type="revolute">
    <parent link="base_link"/>
    <child link="f3_finger_base"/>
    <origin rpy="0 3.14 0" xyz="-0.028 0.02 -0.019"/>
    <limit effort="1000.0" lower="-0.785" upper="0.087" velocity="0.5"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="f3_finger_topjoint" type="revolute">
    <parent link="f3_finger_base"/>
    <child link="f3_finger_top"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0 0 0" xyz="0 0.056 0"/>
    <limit effort="1000.0" lower="1.431" upper="0.489" velocity="0.5"/>
  </joint>
</robot>
